<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>860</width>
    <height>767</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="RVWidget" name="widgetRV">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>300</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Utiliser la souris pour bouger le cube</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="animationButton">
          <property name="toolTip">
           <string notr="true">Activer l'animation</string>
          </property>
          <property name="text">
           <string>Animation</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Réglages caméra</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Ouverture focale</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDial" name="fov">
             <property name="toolTip">
              <string>Change la focale de la caméra</string>
             </property>
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>120</number>
             </property>
             <property name="pageStep">
              <number>5</number>
             </property>
             <property name="value">
              <number>45</number>
             </property>
             <property name="tracking">
              <bool>true</bool>
             </property>
             <property name="notchTarget">
              <double>5.000000000000000</double>
             </property>
             <property name="notchesVisible">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Réglages objet</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QCheckBox" name="checkBox">
             <property name="text">
              <string>Fil de fer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Opacité</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="opacitySlider">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Modfie l'opacité</string>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBothSides</enum>
             </property>
             <property name="tickInterval">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Taille</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>400</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Source Lumineuse</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QSpinBox" name="xLight">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <number>-99</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="yLight">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <number>-99</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="zLight">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <number>-99</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ambientButton">
             <property name="styleSheet">
              <string notr="true">background-color: rgb(128, 128, 128);
border: 1px solid black;
border-radius: 5px;</string>
             </property>
             <property name="text">
              <string>Couleur Ambiante</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="diffuseButton">
             <property name="styleSheet">
              <string notr="true">background-color: rgb(252, 253, 255);
border: 1px solid black;
border-radius: 5px;</string>
             </property>
             <property name="text">
              <string>Couleur Diffuse</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="specularButton">
             <property name="styleSheet">
              <string notr="true">background-color: rgb(255, 255, 127);
border: 1px solid black;
border-radius: 5px;</string>
             </property>
             <property name="text">
              <string>Couleur Spéculaire</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Coeff. Réflex. Spéculaire</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider_2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>75</number>
             </property>
             <property name="value">
              <number>15</number>
             </property>
             <property name="sliderPosition">
              <number>15</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>860</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RVWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>rvwidget.h</header>
   <slots>
    <slot>startAnimation()</slot>
    <slot>changeFov(int)</slot>
    <slot>changeOpacity(int)</slot>
    <slot>changeWireFrame(bool)</slot>
    <slot>changeCulling(bool)</slot>
    <slot>changeScale(int)</slot>
    <slot>changeSaturation(int)</slot>
    <slot>changeCameraType(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>animationButton</sender>
   <signal>clicked()</signal>
   <receiver>widgetRV</receiver>
   <slot>startAnimation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fov</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeFov(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>802</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>opacitySlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeOpacity(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>819</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeWireFrame(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>819</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>572</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeScale(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>590</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ambientButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeAmbientLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>622</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>566</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>diffuseButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeDiffuseLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>646</y>
    </hint>
    <hint type="destinationlabel">
     <x>736</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>specularButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeSpecularLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>670</y>
    </hint>
    <hint type="destinationlabel">
     <x>771</x>
     <y>568</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xLight</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeXLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yLight</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeYLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>570</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>431</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zLight</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeZLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>598</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>457</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeSpecCoeff(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>720</y>
    </hint>
    <hint type="destinationlabel">
     <x>648</x>
     <y>549</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changeAmbientLight()</slot>
  <slot>changeDiffuseLight()</slot>
  <slot>changeSpecularLight()</slot>
  <slot>changeXLight()</slot>
  <slot>changeYLight()</slot>
  <slot>changeZLight()</slot>
  <slot>changeSpecCoeff(int)</slot>
 </slots>
</ui>
